### PR TITLE
deprecate using referrer as user scope for lti

### PR DESCRIPTION
**Pull request by bracken**

### CODE REVIEW

File: app/controllers/lti_base_controller.rb [hunk 1]
```diff
@@ -70,7 +81,7 @@ def do_lti
 
         @external_identifier = @user.external_identifiers.create!(
                 identifier: @identifier,
-                provider: @lti_provider,
+                provider: lti_provider || deprecated_lti_provider,
```
Comment (levionessa) [inline app/controllers/lti_base_controller.rb]: Thank you for the naming! Super helpful!

File: app/controllers/lti_base_controller.rb [hunk 2]
```diff
@@ -5,13 +5,16 @@ class LtiBaseController < ApplicationController
   #
   # LTI related functionality:
   #
-  def lti_provider
-    params[:tool_consumer_instance_guid] ||
+  def deprecated_lti_provider
```
Comment (levionessa) [inline app/controllers/lti_base_controller.rb]: Would it be worth adding a warn-level log statement to this method to say it's deprecated when it gets used? Might help identify when it's no longer in use.

**Comment by levionessa** [comment]
:+1: :shipit:

File: app/controllers/lti_base_controller.rb (hunk 2 above)
Comment (bracken) [reply app/controllers/lti_base_controller.rb]: yeah, that seems like a good idea.

**Pull request merged**